src/operators/Remap/avtRemapFilter.C

```diff
@@ -131,447 +138,528 @@ avtRemapFilter::SetAtts(const AttributeGroup *a)
+        for (int i = 0; i < numChildren; ++i)
+        {
+            if (inTree->ChildIsPresent(i))
+            {
+                TraverseDomainTree(inTree->GetChild(i));
+            }
+            else
+            {
+                debug4 << "Child " << i << " is not present. Skipping." << std::endl;
+            }
+        }
+    }
+}
+
+
+// ****************************************************************************
+//  Method: avtRemapFilter::ClipDomain
+//
+//  Purpose:
+//      Clip the input domain against the output rectilinear grid.
+//
+//  Arguments:
+//      inLeaf     avtDataTree_p leaf, as determined by TraverseDomainTree.
+//
+//  Programmer: rusu1
+//  Creation:   Wed Apr  3 13:52:34 PDT 2019
+//
+// ****************************************************************************
+
+void
+avtRemapFilter::ClipDomain(avtDataTree_p inLeaf)
+{
+    debug3 << "avtRemapFilter::ClipDomain" << std::endl;
+
+    // --------------------------------------------------- //
+    // --- Convert the avtDataTree_p into a vtkDataSet --- //
+    // --------------------------------------------------- //
+    avtDataRepresentation in_dr = inLeaf->GetDataRepresentation();
+    int domainId = in_dr.GetDomain();
+    vtkDataSet* in_ds = in_dr.GetDataVTK();
+    
+    // If the dataset does not exist, then return.
+    // If the variable does not exist, then return.
+    // If there are no points, then return.
+    // If there are no cells, then return.
+    if (in_ds == NULL || in_ds->GetCellData()->GetArray(vars->GetName()) == NULL ||
+        in_ds->GetNumberOfPoints() == 0 || in_ds->GetNumberOfCells() == 0)
+    {
+        debug4 << "Domain " << domainId << " is invalid." << std::endl;
+        return;
     }
     
-    // Determine the variable type from the attributes
-    RemapAttributes::VariableTypes varType =
-        atts.GetVariableType();
     
-//std::cout << "Added blank vars to rg:" << std::endl;
-//PrintData(rg);
-     
-       
     // ------------------------------------------------------- //
     // --- Calculate volumes of each cell in Original Grid --- //
     // ------------------------------------------------------- //
-    // add the original volume to the in_ds AFTER the variables have already
-    // been added to rg so that we don't add an artificial variable.
     
+    // Add the original volume to the in_ds AFTER the variables have already
+    // been added to rg so that we don't add an artificial variable.
     vtkDoubleArray* avtRemapOriginalVolume = CalculateCellVolumes(in_ds,
             "avtRemapOriginalVolume");
     in_ds->GetCellData()->AddArray(avtRemapOriginalVolume);
-    //in_ds->GetCellData()->CopyFieldOn("avtRemapOriginalVolume"); // Deep copy
-    //avtRemapOriginalVolume->Delete(); // Clean
-    
-std::cout << "In_ds after adding volumes" << std::endl;   
-PrintData(in_ds);
-//in_ds->Print(std::cout);
+    debug5 << "Added cell volumes to in_ds" << std::endl;
```

Comment:
I just noticed that you are modifying the input dataset. That is not our practice. I believe you should create a NewInstance of in_ds and ShallowCopy its contents. @brugger1, can you comment on this?

Reply:
Just pushed up a change for what I believe you're asking. Let me know if I should modify it.